core/variable_map.ts

```diff
@@ -370,12 +363,9 @@ export class VariableMap {
    * @return All of the variable names of all types.
    */
   getAllVariableNames(): string[] {
-    const allNames = [];
-    for (const key in this.variableMap_) {
-      const variables = this.variableMap_[key];
-      for (let i = 0, variable; variable = variables[i]; i++) {
-        allNames.push(variable.name);
-      }
+    let allNames: string[] = [];
+    for (const variables of this.variableMap.values()) {
+      allNames = allNames.concat(variables.map(variable => variable.name));
```

Comment:
we should have access to `flat` as in `return this.variableMap.values().flat().map(variable => variable.name)` now I think.

Reply:
Nice, done!